tianshou/policy/base.py

```diff
@@ -138,9 +137,7 @@ def post_process_fn(
         Typical usage is to update the sampling weight in prioritized
         experience replay. Used in :meth:`update`.
         """
-        if isinstance(buffer, PrioritizedReplayBuffer) and hasattr(
-            batch, "weight"
-        ):
+        if hasattr(buffer, "update_weight") and hasattr(batch, "weight"):
```

Comment:
It would be better to provide a `ReplayBufferBase` with `update_weight` doing nothing by default, instead of relying on attribute checking.

Reply:
Good idea!

Reply:
I do not agree. ``ReplayBufferBase `` does not fit into the concept of ``weight``. This is specific to prioritized replaybuffer.

Reply:
If so, provide a `PrioritizedReplayBufferBase` then. It is the same Logic: never check for attribute existence, but for inheritence.

Reply:
So where should this base class be placed?

Reply:
I think ``PrioritizedReplayBufferBase`` is not necessary when we have just one per buffer.